src/app/classes/classes.component.ts

```diff
@@ -9,19 +9,47 @@ import { SubscriptionService } from '../services/subscription.service';
   styleUrls: ['./classes.component.css'],
 })
 export class ClassesComponent implements OnInit {
-  classes: Class[];
-  selectedClass: Class;
+  allClasses: Class[];
+  searchText = '';
+  displayedClasses: Class[];
+  subscribedClases: string[];
 
   constructor(private subService: SubscriptionService) {}
 
   ngOnInit() {
+    this.subService.getClasses().subscribe((classes) => {
+      (this.allClasses = classes), (this.displayedClasses = this.allClasses);
```

Comment:
I cannot comprehend this line. I don't even know how the compiler interprets it.